classes/form/CustomerForm.php

```diff
@@ -211,12 +213,17 @@ public function submit()
             $clearTextPassword = $this->getValue('password');
             $newPassword = $this->getValue('new_password');
 
-            $ok = $this->customerPersister->save(
-                $this->getCustomer(),
-                $clearTextPassword,
-                $newPassword,
-                $this->passwordRequired
-            );
+            try {
+                $ok = $this->customerPersister->save(
+                    $this->getCustomer(),
+                    $clearTextPassword,
+                    $newPassword,
+                    $this->passwordRequired
+                );
+            } catch (PrestaShopException $e) {
+                $this->errors[] = $this->translator->trans('Could not update your information, please check your data.', [], 'Shop.Notifications.Error');
```

Comment:
In Form context, errors are in format `array<int, array>` I'm not sure it's  `array<int, string>`

Reply:
After manual testing

The errors array key needs to be a string and its value an array with the error msg as string

if the key is a field name, it will be displayed in front of the field. if it's empty string it's a global error

That explains why it's instantiated like https://github.com/PrestaShop/PrestaShop/blob/1.7.8.x/classes/form/AbstractForm.php#L41

Reply:
It's clearly a weird syntax, but anyway :sweat_smile: